test/mpi/threads/perf/mt_rma_msgrate.c

```diff
@@ -89,8 +100,16 @@ MTEST_THREAD_RETURN_TYPE thread_fn(void *arg)
                                        1, 0, MESSAGE_SIZE, MPI_CHAR, MPI_NO_OP, my_win);
                 }
             }
+            if (target_type == TARGET_PASSIVE) {
+                MPI_Win_flush(1, my_win);
+            }
         }
-        MPI_Win_fence(0, my_win);
+        if (target_type == TARGET_ACTIVE) {
+            MPI_Win_fence(0, my_win);
+        }
+    }
+    if (target_type == TARGET_PASSIVE) {
+        MPI_Win_unlock_all(my_win);
     }
```

Comment:
these branches are a bit hard to look at. i welcome any suggestions to improve.

Reply:
How about wrapping the `if-` inside macros? For example 
```
MY_EPOCH_START
for (win_post_i ...) {
    <issue ops>
    MY_WIN_FLUSH
}
MY_EPOCH_END
```

Reply:
> How about wrapping the `if-` inside macros? For example
> 
> ```
> MY_EPOCH_START
> for (win_post_i ...) {
>     <issue ops>
>     MY_WIN_FLUSH
> }
> MY_EPOCH_END
> ```

Thanks, this should be an improvement.